core/src/main/java/com/alibaba/nacos/core/auth/AuthFilter.java

```diff
@@ -73,10 +73,25 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         HttpServletRequest req = (HttpServletRequest) request;
         HttpServletResponse resp = (HttpServletResponse) response;
         
-        String userAgent = WebUtils.getUserAgent(req);
-        
-        if (StringUtils.startsWith(userAgent, Constants.NACOS_SERVER_HEADER)) {
-            chain.doFilter(request, response);
+        if (authConfigs.isEnableUserAgentAuthWhite()) {
+            String userAgent = WebUtils.getUserAgent(req);
+            if (StringUtils.startsWith(userAgent, Constants.NACOS_SERVER_HEADER)) {
+                chain.doFilter(request, response);
+                return;
+            }
+        } else if (StringUtils.isNotBlank(authConfigs.getServerIdentityKey()) && StringUtils
+                .isNotBlank(authConfigs.getServerIdentityValue())) {
+            String serverIdentity = req.getHeader(authConfigs.getServerIdentityKey());
+            if (authConfigs.getServerIdentityValue().equals(serverIdentity)) {
```

Comment:
Please note that comparing credential with `equals` makes your application vulnerable to [timing attack](https://en.wikipedia.org/wiki/Timing_attack)

Reply:
How to solve it? Can you provide some advise for this?

Reply:
https://github.com/playframework/play1/blob/master/framework/src/play/mvc/CookieDataCodec.java#L82

Reply:
In my opinion,  there is enough for equals. reason follow:

1. Nacos is Internal IDC application, we recommend users use Nacos with internal network not public network, If hacker want to timing attack, the network has been controlled by hacker. And Cost much time to analyze package between nacos server.
2. The key and value is defined by configuration file, as the first item said, if the internal network is in controlled by hacker. they can find the key and value by configuration file directly.

I'm not sure whether the timing attack is send a long string to hold equals resource. If so, I think we has no ability to solve all `public application security problem` for an internal network application.

Reply:
@KomachiSion FYI maybe you should use [`MessageDigest.isEquals` ](https://docs.oracle.com/javase/7/docs/api/java/security/MessageDigest.html)